lib/chef/provider/launchd.rb

```diff
@@ -0,0 +1,198 @@
+#
+# Author:: Mike Dodge (<mikedodge04@gmail.com>)
+# Copyright:: Copyright (c) 2015 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/provider'
+require 'chef/resource'
+require 'chef/resource/file'
+require 'chef/resource/cookbook_file'
+require 'chef/resource/macosx_service'
+
+class Chef
+  class Provider
+    class Launchd < Chef::Provider
+
+      def load_current_resource
+        @current_resource = Chef::Resource::Launchd.new(@new_resource.name)
+        @name = @new_resource.name
+        @label = @new_resource.label ? @new_resource.label : @name
+        @backup = @new_resource.backup
+        @cookbook = @new_resource.cookbook
+        @group = @new_resource.group
+        @mode = @new_resource.mode
+        @owner = @new_resource.owner
+        @path = @new_resource.path
+        @source = @new_resource.source
```

Comment:
you could drop all the `@`signs off of `@new_resource` here.

it also looks more like you want to use `def_delegators` here to forward `#path` to `new_resource.path`.